name.abuchen.portfolio/src/name/abuchen/portfolio/model/ClientFactory.java

```diff
@@ -201,12 +203,40 @@ public Decryptor(String method, char[] password)
             this.password = password;
             this.keyLength = "AES256".equals(method) ? AES256_KEYLENGTH : AES128_KEYLENGTH; //$NON-NLS-1$
         }
+        
+        /**
+         * Reads all data in the given {@link InputStream} ignoring
+         * {@link BadPaddingException}s.
+         * 
+         * @param is
+         *            {@link InputStream} (is closed when this function returns)
+         * @throws IOException
+         */
+        private static void readStreamUntilEnd(InputStream is) throws IOException
+        {
+            try
+            {
+                // use an appropriate buffer (64 KB here) and loop until end.
+                byte[] data = new byte[65536];
+                while (is.read(data) != -1)
+                {}
+                // explicitly close the stream to force bad padding exceptions
+                // to occur inside this function
+                is.close();
```

Comment:
Ich habe es gerade mal probiert: Exceptions aus try with Block werden auch in dem catch block behandelt. D.h. es braucht eigentlich kein explizites #close hier.

Reply:
Hier war die Idee das Schließen zu erzwingen, da es sonst [hier](https://github.com/buchen/portfolio/pull/845/commits/61957303f7a5d0944c7eba2ddb73d4248414449e#diff-688b3898a7317c9b261c210d8f133750L279) passieren könnte und dann wieder die Exception durchrutscht.
Aber siehe nächster Kommentar.